client/src/components/PostModal.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { tweetNewMessage } from '../redux/actions/twitterActions';
+
+const PostModal = () => {
+  const [postMessage, setPostMessage] = useState('');
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      {/* the button to open tweet modal */}
+      <button
+        type="button"
+        className="btn btn-primary"
+        data-bs-toggle="modal"
+        data-bs-target="#postModal"
+        onClick={() => setPostMessage('')}
+      >
+        Create Tweet
+      </button>
+      {/* the actual modal */}
+      <div className="modal" tabIndex="-1" id="postModal">
```

Comment:
This modal can also be a component on it's own as well and pass functions as props like onSubmit onClose

Reply:
created ModalBody component